man/madlib.kmeans.Rd

```diff
@@ -0,0 +1,142 @@
+\name{madlib.kmeans}
+\alias{madlib.kmeans}
+
+\title{Wrapper for MADlib's Kmeans clustering function}
+
+
+\description{
+  The wrapper function for MADlib's kmeans clustering [1]. Clustering refers
+  to the problem of partitioning a set of objects according to some
+  problem-dependent measure of similarity. Each centroid represents a cluster
+  that consists of all points to which this centroid is closest. The
+  computation is parallelized by MADlib if the connected database is
+  Greenplum/HAWQ database.
+
+}
+
+
+\usage{
+
+madlib.kmeans(
+  x, centers, iter.max = 10, nstart = 1, algorithm = "Lloyd",
+  pid, expr, expr.centroid = NULL,
+  fn.dist = "squared_dist_norm2", agg.centroid = "avg", min.frac = 0.001,
+  kmeanspp = FALSE, seeding.sample.ratio=1.0, ...)
+
+}
+
+\arguments{
+
+  \item{x}{
+    An object of \code{db.obj} class. Currently, this parameter is
+    mandatory. If it is an object of class \code{db.Rquery} or
+    \code{db.view}, a temporary table will be created, and further
+    computation will be done on the temporary table. After the computation,
+    the temporary will be dropped from the corresponding database.
+  }
+
+  \item{centers}{
+
+    A number, a matrix or a string which indicates how the centers are
+    formed by the function. If it is a number, this sets the number of
+    target centroids and the random (or kmeans++) seeding method is
+    used. Otherwise, this parameters is used for initial centers. If
+    it is a matrix, its rows will denote the initial centroid
+    coordinates. Otherwise this parameter will contain the name of a
```

Comment:
Now that I think about, IMO we shouldn't support the database name input but force the user to create a db.table even if it's done during the function call.